src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -746,30 +746,41 @@ static async Task ExecuteAwaited(ValueTask<string> task, HttpContext httpContext
 
             if (task.IsCompletedSuccessfully)
             {
-                return httpContext.Response.WriteAsync(task.GetAwaiter().GetResult());
+                return httpContext.Response.WriteAsync(task.GetAwaiter().GetResult()!);
             }
 
-            return ExecuteAwaited(task, httpContext);
+            return ExecuteAwaited(task!, httpContext);
         }
 
-        private static Task ExecuteValueTaskResult<T>(ValueTask<T> task, HttpContext httpContext) where T : IResult
+        private static Task ExecuteValueTaskResult<T>(ValueTask<T?> task, HttpContext httpContext) where T : IResult
         {
+            EnsureRequestTaskOfNotNull(task.AsTask());
```

Comment:
Avoid call `.AsTask()` especially if it might be a completed `ValueTask<T>` because it allocates.

Reply:
👍 . I proposed **`EnsureRequestResultNotNull(await task)`** and **`EnsureRequestResultNotNull(task.GetAwaiter().GetResult())`**